docs/api-docs/apps/tutorials/sample-app-next/05-step-3-integrate-api.md

```diff
@@ -0,0 +1,920 @@
+```
+
+<div class="HubBlock--callout">
+<div class="CalloutBlock--info">
+<div class="HubBlock-content">
+
+> ### Note
+> In the development mode, every time you modify your environment variables, make sure to restart the process (`npm run dev`) to capture the changes.
+
+</div>
+</div>
+</div>
+
+## Set up the MySQL database
+
+[MySQL](https://www.mysql.com/) is a relational database management system.
+
+### Configure MySQL
+
+1. In the `dbs` folder, create a `mysql.ts` file.
+
+2. At the top of the file, add the following imports:
+
+```js
+import * as mysql from 'mysql';
+import { promisify } from 'util';
+import { SessionProps, StoreData } from '../../types';
+```
+
+3. Add the MySQL config and initialization logic.
+
+```js
+const connection = mysql.createConnection(process.env.CLEARDB_DATABASE_URL);
+const query = promisify(connection.query.bind(connection));
+
+export async function setUser({ context, user }: SessionProps) {
+    if (!user) return null;
+
+    const { email, id, username } = user;
+    const storeHash = context?.split('/')[1] || '';
+
+    const userData = { email, userId: id, storeHash, username };
+
+    await query('REPLACE INTO users SET ?', userData);
+}
+
+export async function setStore(session: SessionProps) {
+    const { access_token: accessToken, context, scope } = session;
+    // Only set on app install or update
+    if (!accessToken || !scope) return null;
+
+    const storeHash = context?.split('/')[1] || '';
+
+    const storeData: StoreData = { accessToken, scope, storeHash };
+    await query('REPLACE INTO stores SET ?', storeData);
+}
+
+export async function getStore() {
+    const results = await query('SELECT * from stores limit 1');
+
+    return results.length ? results[0] : null;
+}
+
+export async function getStoreToken(storeHash: string) {
+    if (!storeHash) return null;
+
+    const results = await query('SELECT accessToken from stores limit 1');
+
+    return results.length ? results[0].accessToken : null;
+}
+
+export async function deleteStore({ store_hash: storeHash }: SessionProps) {
+    await query('DELETE FROM stores WHERE storeHash = ?', storeHash);
+}
+```
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-3-add-database/lib/dbs/mysql.ts)
+
+### Update environment variables
+
+1. In the `.env` file, specify the database type.
+
+```
+DB_TYPE=mysql
+```
+
+2. Enter your MySQL database config keys (host, database, username, password, and port).
+
+```
+MYSQL_HOST={mysql host}
+MYSQL_DATABASE={mysql domain}
+MYSQL_USERNAME={mysql username}
+MYSQL_PASSWORD={mysql password}
+MYSQL_PORT={mysql port}
+```
+
+## Setup the db lib page
+
+2. In the `lib` folder, create a `db.ts` file.
+
+3. Open the `db.ts` file and add the `Db` import at the top of the file.
+
+```js
+import { Db } from '../types'
+```
+
+4. Add the switch expression to determine which database code to execute.
+
+```js
+const { DB_TYPE } = process.env;
+
+let db: Db;
+
+switch (DB_TYPE) {
```

Comment:
🍹 I'm going back-and-forth on this one.  Given that we've broken the DBs into their own section, should we extend those sections by only showing how to config `/lib/` for that specific db?  In other words, don't show this switch in the tutorial, but rather move `Update the auth lib page`, part 8 to each db section. @christensenep what're your thoughts?

e.g. 
For Firebase: `Add the setSession, getSession, and removeSession functions:`
```tsx
import * as fire from './dbs/firebase';

export async function setSession(session: SessionProps) {
    fire.setUser(session);
    fire.setStore(session);
}

export async function getSession({ query: { context = '' } }: NextApiRequest) {
    if (typeof context !== 'string') return;
    const decodedContext = decodePayload(context)?.context;
    const accessToken = await fire.getStoreToken(decodedContext);

    return { accessToken, storeHash: decodedContext };
}

export async function removeSession(res: NextApiResponse, session: SessionProps) {
    await fire.deleteStore(session);
}
```